### PR TITLE
fix(website): press the Report an Issue button makes error if eslintrc does not include rules

### LINKED ISSUE #10365: Bug(website): clicking playground's "Report as Issue" throws exception when "rules" is omitted from eslintrc
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

website

### Playground Link

http://typescript-eslint.io/play/#ts=5.6.2&showAST=es&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3gMyUgXPJjAJYoDm8APvAK4qjZkjADcAUO8ukhCAHQQk5ABS4k8APyS6DEExQsAlKyA&eslintrc=N4KABGBECmAeAu0B2ATAzpAXGA2uCUADgDYCuA5gJZKYAC8AnodGgMYBOlh8AtC8dXgB6NPE6tejZj1YALaKwDW0FJHwBdEAF8gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
n/a
```

### ESLint Config

```javascript
{
  "extends": [
    "plugin:@typescript-eslint/strict-type-checked"
  ]
}
```

### tsconfig

```jsonc
n/a
```

### Expected Result

I expect the "Report as Issue" button to bring me to a pre-filled github issue template

### Actual Result

An exception is thrown, visible in console

### Additional Info

Running this locally, even though the stack trace isn't minified, it still isn't terribly helpful
```text
ERROR
Cannot convert undefined or null to object
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at createMarkdownParams (webpack-internal:///./src/components/lib/markdown.ts:11:152)
    at eval (webpack-internal:///./src/components/OptionsSelector.tsx:24:574)
    at HTMLUnknownElement.callCallback (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:4164:14)
    at Object.invokeGuardedCallbackDev (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:4213:16)
    at invokeGuardedCallback (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:4277:31)
    at invokeGuardedCallbackAndCatchFirstError (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:4291:25)
    at executeDispatch (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:9041:3)
    at processDispatchQueueItemsInOrder (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:9073:7)
    at processDispatchQueue (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:9086:5)
```

### Versions

main, 8.15.0

### PR DESCRIPTION
**Pull request by developer-bandi**
## PR Checklist

- [x] Addresses an existing open issue: fixes #10365 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Fixed an error that occurred when pressing the Report an Issue button when eslintrc does not include rules.

### CODE REVIEW

**Review by JoshuaKGoldberg — APPROVED** [review]
Alan Shane Lewis saying 'absolute perfection' on The Great Canadian Baking Show

**Review by kirkwaiblinger — APPROVED** [review]

**Comment by kirkwaiblinger** [comment]
Nice, thanks! 🙂

File: packages/website/src/components/lib/markdown.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export function createMarkdown(state: ConfigModel): string {
  */
 export function createMarkdownParams(state: ConfigModel): string {
   const { rules } = parseESLintRC(state.eslintrc);
-  const ruleKeys = Object.keys(rules);
+  const ruleKeys = Object.keys(rules ?? {});
```
Comment (kirkwaiblinger) [inline packages/website/src/components/lib/markdown.ts:57]: Oh, I guess you'll have to fix the lint error that complains about this not being possibly-nullish

**Review by kirkwaiblinger — APPROVED** [review]
Thanks!

**Pull request merged**